test/vault_helpers.go

```diff
@@ -487,6 +340,60 @@ func runVaultEC2AuthTest(t *testing.T, packerBuildName string) {
 	})
 }
 
+func setupAmi(t *testing.T, examplesDir string, packerBuildName string) {
+	awsRegion := aws.GetRandomRegion(t, nil, nil)
+	test_structure.SaveString(t, examplesDir, SAVED_AWS_REGION, awsRegion)
+
+	tlsCert := generateSelfSignedTlsCert(t)
+	saveTlsCert(t, examplesDir, tlsCert)
+
+	amiId := buildAmi(t, AMI_EXAMPLE_PATH, packerBuildName, tlsCert, awsRegion)
+	test_structure.SaveAmiId(t, examplesDir, amiId)
+}
+
+func teardownResources(t *testing.T, examplesDir string) {
+	terraformOptions := test_structure.LoadTerraformOptions(t, examplesDir)
+	terraform.Destroy(t, terraformOptions)
+
+	amiId := test_structure.LoadAmiId(t, examplesDir)
+	awsRegion := test_structure.LoadString(t, examplesDir, SAVED_AWS_REGION)
+	aws.DeleteAmi(t, awsRegion, amiId)
+
+	keyPair := test_structure.LoadEc2KeyPair(t, examplesDir)
+	aws.DeleteEC2KeyPair(t, keyPair)
+
+	tlsCert := loadTlsCert(t, examplesDir)
+	cleanupTlsCertFiles(tlsCert)
+}
+
+func deployCluster(t *testing.T, examplesDir string, uniqueId string, terraformVars map[string]interface{}) {
+	amiId := test_structure.LoadAmiId(t, examplesDir)
+	awsRegion := test_structure.LoadString(t, examplesDir, SAVED_AWS_REGION)
+
+	keyPair := aws.CreateAndImportEC2KeyPair(t, awsRegion, uniqueId)
+	test_structure.SaveEc2KeyPair(t, examplesDir, keyPair)
+
+	for key, value := range map[string]interface{}{
+		VAR_AMI_ID:                 amiId,
+		VAR_VAULT_CLUSTER_NAME:     fmt.Sprintf("vault-test-%s", uniqueId),
+		VAR_CONSUL_CLUSTER_NAME:    fmt.Sprintf("consul-test-%s", uniqueId),
+		VAR_CONSUL_CLUSTER_TAG_KEY: fmt.Sprintf("consul-test-%s", uniqueId),
+		VAR_SSH_KEY_NAME:           keyPair.Name,
+	} {
+		terraformVars[key] = value
+	}
```

Comment:
This is a clever but somewhat unusual way to just merge two maps. Can you make this more clear by writing your own mini `merge` function that's more generic in nature and then call that? Should make the code more readable.

Reply:
sure!